### PR TITLE
CSW - Add reloading from nearSupplies

### PR DESCRIPTION
**Pull request by LinkIsGrim**
**When merged this pull request will:**
- Add ability for players to reload from nearby vehicles, friendly AI with ammo, boxes, ground holders, and dropped backpacks.
- Make AI reload upon entering an empty CSW.
- Move AI CSW reloading to its own function.
- Change a config typeOf lookup to configOf.
- Fix AI CSW reloading with magazines different than currently loaded magazine.
- Fix some function headers.
- Stop AI from being able to reload CSW's from players or enemies' inventory.
- Change a CBA namespace in CSW to native hashMap.
- Micro-optimizations in reload functions to avoid solving config lookups within loops.
- Improve `common_fnc_removeSpecificMagazine`'s interaction with WeaponHolders.
- Capitalize setting titles.
- Fix noun order in some portuguese strings.
- Add setting to change advanced assembly's ammo returning to use a holder item with vanilla ammo box model instead of a really messy GroundWeaponHolder. Holder is draggable/carryable, but items can't be inserted into it by players. It is still possible to pull magazines from it. Subjective change, but I've seen a fair few communities not use the feature because of this.

### IMPORTANT

- [ ] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3mod.com/).
- [x] [Development Guidelines](https://ace3mod.com/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

**Comment by Beeperz** [comment]
Please approve it, need so much!

File: addons/csw/functions/fnc_reload_loadMagazine.sqf [hunk 1]
```diff
@@ -1,25 +1,26 @@
...
- * [cursorTarget, [0], "ACE_csw_100Rnd_127x99_mag_red", player] call ace_csw_fnc_reload_loadMagazine
+ * [cursorTarget, [0], "ACE_csw_100Rnd_127x99_mag_red", player, player] call ace_csw_fnc_reload_loadMagazine
  *
  * Public: No
  */
 
-params ["_vehicle", "_turret", "_carryMag", "_unit"];
-TRACE_4("loadMagazine",_vehicle,_turret,_carryMag,_unit);
+params ["_vehicle", "_turret", "_carryMag", "_magSource", "_unit"];
+TRACE_4("loadMagazine",_vehicle,_turret,_carryMag,_magSource,_unit);
```
Comment (PabstMirror) [inline addons/csw/functions/fnc_reload_loadMagazine.sqf:23]:
```suggestion
TRACE_5("loadMagazine",_vehicle,_turret,_carryMag,_magSource,_unit);
```

File: addons/csw/initSettings.sqf [hunk 2]
```diff
@@ -20,6 +20,16 @@ private _categoryArray = [format ["ACE %1", localize LSTRING(DisplayName)]];
     true // Needs mission restart
 ] call CBA_fnc_addSetting;
 
+[
+    QGVAR(handleExtraMagazinesType), "LIST",
```
Comment (PabstMirror) [inline addons/csw/initSettings.sqf:24]: we could merge this with the `handleExtraMagazines` setting
0 - off
1 - ground
2 - box

but I think it's fine as is

File: addons/csw/functions/fnc_ai_reload.sqf [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    private _cswMagazines  = [];
+    {
+        _cswMagazines pushBackUnique _x;
+    } forEach ((magazineCargo _xSource) select {isClass (_cfgMagGroups >> _x)});
+    TRACE_2("",_xSource,_cswMagazines);
+
+    private _compatibleMags = [_weapon] call CBA_fnc_compatibleMagazines;
+    if (_magazine != "") then {
+        systemChat str _magazine;
```
Comment (PabstMirror) [inline addons/csw/functions/fnc_ai_reload.sqf:44]:
```suggestion
```

File: addons/csw/initSettings.sqf (hunk 2 above)
Comment (Salluci) [reply addons/csw/initSettings.sqf:24]: I'd rather not invalidate the current setting.

**Review by PabstMirror — APPROVED** [review]

**Pull request merged**